scripts/check-release.js

```diff
@@ -0,0 +1,103 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { version: releaseVersion } = require('../lerna.json')
+const { findBrowserSdkPackageJsonFiles, printLog, logAndExit, executeCommand } = require('./utils')
+
+async function main() {
+  await checkGitTag()
+  await checkBrowserSdkPackageJsonFiles()
+
+  printLog('Release check done.')
+}
+
+async function checkGitTag() {
+  printLog('Checking release version tag is on HEAD')
+  const headRef = await executeCommand('git rev-parse HEAD')
+  let tagRef
+  try {
+    tagRef = await executeCommand(`git rev-list -n 1 v${releaseVersion} --`)
+  } catch (error) {
+    throw new Error(`Failed to find git tag reference: ${error}`)
+  }
+  if (tagRef !== headRef) {
+    throw new Error('Git tag not on HEAD')
+  }
+}
+
+async function checkBrowserSdkPackageJsonFiles() {
+  const packageJsonFiles = await findBrowserSdkPackageJsonFiles()
+
+  printLog(
+    `Checking package.json files:
+   - ${packageJsonFiles.map((packageJsonFile) => packageJsonFile.relativePath).join('\n   - ')}
+`
+  )
+
+  for (const packageJsonFile of packageJsonFiles) {
+    checkPackageJsonFile(packageJsonFile)
+  }
+}
+
+function checkPackageJsonFile(packageJsonFile) {
+  if (
+    isBrowserSdkPublicPackageName(packageJsonFile.content.name) &&
+    packageJsonFile.content.version !== releaseVersion
+  ) {
+    throw new Error(
+      `Invalid version for ${packageJsonFile.relativePath}: expected ${releaseVersion}, got ${packageJsonFile.content.version}`
+    )
+  }
+  checkPackageDependencyVersions(packageJsonFile)
+  checkPackageJsonEntryPoints(packageJsonFile)
+}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏in order to have a consistent level of abstraction and avoid one wrapping function

```suggestion
  for (const packageJsonFile of packageJsonFiles) {
    checkPackageJsonVersions(packageJsonFile)
    checkPackageDependencyVersions(packageJsonFile)
    checkPackageJsonEntryPoints(packageJsonFile)
  }
}

function checkPackageJsonVersions(packageJsonFile) {
  if (
    isBrowserSdkPublicPackageName(packageJsonFile.content.name) &&
    packageJsonFile.content.version !== releaseVersion
  ) {
    throw new Error(
      `Invalid version for ${packageJsonFile.relativePath}: expected ${releaseVersion}, got ${packageJsonFile.content.version}`
    )
  }
}
```